### PR TITLE
feat: declarative Cloud Run workflow

### PR DESCRIPTION
**Pull request by cstanger**
This PR will add an additional example of a cloud run deploy workflow using a declarative approach with YAML Service specification.

### CODE REVIEW

File: workflows/deploy-cloudrun/cloudrun-declarative.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  deploy:
+    # Add 'id-token' with the intended permissions for workload identity federation
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (sethvargo) [inline workflows/deploy-cloudrun/cloudrun-declarative.yml:69]: This should be v3 now

File: workflows/deploy-cloudrun/cloudrun-declarative.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      #   uses: 'google-github-actions/auth@v0'
+      #   with:
+      #     credentials_json: '${{ secrets.GCP_CREDENTIALS }}''
+
+      # BEGIN - Docker auth and build (NOTE: If you already have a container image, these Docker steps can be omitted)
+
+      # Authenticate Docker to Google Cloud Artifact Registry
+      - name: Docker Auth
+        id: docker-auth
+        uses: 'docker/login-action@v1'
```
Comment (sethvargo) [inline workflows/deploy-cloudrun/cloudrun-declarative.yml:91]: v2 now

File: workflows/deploy-cloudrun/cloudrun-declarative.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        id: deploy
+        uses: google-github-actions/deploy-cloudrun@v0
+        with:
+          service: ${{ env.SERVICE }}
+          region: ${{ env.REGION }}
+          metadata: service.yaml
+
+      # If required, use the Cloud Run url output in later steps
+      - name: Show Output
+        run: echo ${{ steps.deploy.outputs.url }}
```
Comment (sethvargo) [inline workflows/deploy-cloudrun/cloudrun-declarative.yml:123]: Nit: need newline EOF

File: workflows/deploy-cloudrun/service.template.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    metadata:
+      annotations:
+        autoscaling.knative.dev/maxScale: '100'
+    spec:
+      containerConcurrency: 80
+      containers:
+      - image: $IMAGE
+  traffic:
+  - percent: 100
+    latestRevision: true
```
Comment (sethvargo) [inline workflows/deploy-cloudrun/service.template.yaml:16]: Newline

File: workflows/deploy-cloudrun/service.template.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: $SERVICE
```
Comment (sethvargo) [inline workflows/deploy-cloudrun/service.template.yaml:4]: The Google style guide suggests always using the curly-brace format:

```suggestion
  name: ${SERVICE}
```

File: workflows/deploy-cloudrun/service.template.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+spec:
+  template:
+    metadata:
+      annotations:
+        autoscaling.knative.dev/maxScale: '100'
+    spec:
+      containerConcurrency: 80
+      containers:
+      - image: $IMAGE
```
Comment (sethvargo) [inline workflows/deploy-cloudrun/service.template.yaml:13]:
```suggestion
      - image: ${IMAGE}
```

File: workflows/deploy-cloudrun/service.template.yaml (hunk 5 above)
Comment (cstanger) [reply workflows/deploy-cloudrun/service.template.yaml]: Thank you for pointing that out... resolved it

**Review by sethvargo — APPROVED** [review]

File: README.md [hunk 7]
```diff
@@ -12,6 +12,7 @@ This repository holds several references to example workflows and demonstrates h
 | ------------------------------------------------------------ | ------------------------- | ---------------- |
 |[cloudrun-docker](workflows/deploy-cloudrun/cloudrun-docker.yml) | ✅ | Build a Docker container, publish it to Google Artifact Registry, and deploy to Google Cloud Run. |
 |[cloudrun-source](workflows/deploy-cloudrun/cloudrun-source.yml) | ✅ | Deploy to Google Cloud Run directly from source. |
+|[cloudrun-declarative](workflows/deploy-cloudrun/cloudrun-declarative.yml) | ✅ | Build a Docker container, publish it to Google Artifact Registry, and deploy to Google Cloud Run using a [declarative YAML Service specification (KRM)](https://cloud.google.com/sdk/gcloud/reference/run/services/replace).  |
```
Comment (verbanicm) [inline README.md:15]: Did you manually add this? Can you delete this line and run `go run scripts/generate.go readme`?

Comment (cstanger) [reply README.md]: yes I created that line manually - resolved by running the script

**Review by verbanicm — APPROVED** [review]

**Pull request merged**